### PR TITLE
Jetpack: remove Personal plan from plans table

### PR DESCRIPTION
**Pull request by monsieur-z**
### Changes proposed in this Pull Request

This PR removes the Jetpack Personal plan from the plans table, unless the customer already owns it. The table appears in 2 places: in the _Plans_ section and Jetpack connect.

### Testing instructions

1. Download the PR
2. Select a self-hosted Jetpack site with a Free plan
3. Go to the _Plans_ section (`/plans/:site/`), check that the Personal plan has been removed, and that the Premium plan contains the "Daily Backup",  "Spam Protection" and "All Free features" features (for both monthly and yearly plans)
4. Go to `/jetpack/connect/plans/yearly/:site/` and do the same check
5. Repeat with a site that has the Personal plan. Check that the Personal plan is present in the pricing table and that the Premium features have not been updated.

### Plans
Site with Personal plan
Screen Shot 2020-07-28 at 4 07 15 PM

Site with other plan
Screen Shot 2020-07-28 at 4 06 39 PM

### Jetpack connect
Screen Shot 2020-07-21 at 12 23 18 PM

### CODE REVIEW

File: client/lib/plans/plans-list.js [hunk 1]
```diff
@@ -719,6 +719,8 @@ export const PLANS_LIST = {
 			] ),
 		getSignupFeatures: () =>
```
Comment (robertf4) [inline client/lib/plans/plans-list.js:720]: This should have `FEATURE_ALL_PERSONAL_FEATURES_JETPACK` replaced with `FEATURE_ALL_FREE_FEATURES_JETPACK` when the personal-plan isn't enabled.

File: client/lib/plans/plans-list.js [hunk 2]
```diff
@@ -788,6 +790,8 @@ export const PLANS_LIST = {
...
```
Comment (robertf4) [inline client/lib/plans/plans-list.js:791]: This should have `FEATURE_ALL_PERSONAL_FEATURES_JETPACK` replaced with `FEATURE_ALL_FREE_FEATURES_JETPACK` when the personal-plan isn't enabled.

File: client/lib/plans/plans-list.js [hunk 3]
```diff
@@ -719,12 +719,14 @@ export const PLANS_LIST = {
...
 			compact( [
+				! isEnabled( 'plans/personal-plan' ) && constants.FEATURE_OFFSITE_BACKUP_VAULTPRESS_DAILY,
+				! isEnabled( 'plans/personal-plan' ) && constants.FEATURE_SPAM_AKISMET_PLUS,
 				constants.FEATURE_MALWARE_SCANNING_DAILY,
 				constants.FEATURE_AUTOMATIC_SECURITY_FIXES,
 				constants.FEATURE_VIDEO_UPLOADS_JETPACK_PRO,
 				constants.FEATURE_WORDADS_INSTANT,
 				constants.FEATURE_ADVANCED_SEO,
-				constants.FEATURE_ALL_PERSONAL_FEATURES_JETPACK,
+				constants.FEATURE_ALL_FREE_FEATURES_JETPACK,
```
Comment (robertf4) [inline client/lib/plans/plans-list.js:729]: This change should also depend on `isEnabled( 'plans/personal-plan' )`, something like this:
```suggestion
				isEnabled( 'plans/personal-plan' ) ? constants.FEATURE_ALL_PERSONAL_FEATURES_JETPACK : constants.FEATURE_ALL_FREE_FEATURES_JETPACK,
```

File: client/lib/plans/plans-list.js [hunk 4]
```diff
@@ -788,12 +790,14 @@ export const PLANS_LIST = {
...
```
Comment (robertf4) [inline client/lib/plans/plans-list.js:800]:
```suggestion
				isEnabled( 'plans/personal-plan' ) ? constants.FEATURE_ALL_PERSONAL_FEATURES_JETPACK : constants.FEATURE_ALL_FREE_FEATURES_JETPACK,
```

File: client/my-sites/plan-features/index.jsx [hunk 5]
```diff
@@ -861,13 +861,15 @@ export default connect(
...
-								planFeatures = getPlanFeaturesObject( planConstantObj.getSignupFeatures( abtest ) );
+								planFeatures = getPlanFeaturesObject(
+									planConstantObj.getSignupFeatures( currentPlan )
+								);
 							}
 					}
 				}
 
 				if ( displayJetpackPlans ) {
-					planFeatures = getPlanFeaturesObject( planConstantObj.getSignupFeatures( abtest ) );
```
Comment (monsieur-z) [inline client/my-sites/plan-features/index.jsx:870]: The various definitions of `getSignupFeatures` in `client/lib/plans/plans-list.js` didn't use a parameter.

**Pull request merged**